nats/js/kv.py

```diff
@@ -337,21 +340,41 @@ async def watchall(self, **kwargs) -> KeyWatcher:
         """
         return await self.watch(">", **kwargs)
 
-    async def keys(self, **kwargs) -> List[str]:
+    async def keys(self, filters: List[str] = None, **kwargs) -> List[str]:
         """
-        keys will return a list of the keys from a KeyValue store.
+        Returns a list of the keys from a KeyValue store.
+        Optionally filters the keys based on the provided filter list.
         """
         watcher = await self.watchall(
             ignore_deletes=True,
             meta_only=True,
         )
         keys = []
 
+        # Check consumer info and make sure filters are applied correctly
+        try:
+            consumer_info = await watcher._sub.consumer_info()
+            if consumer_info and filters:
+                # If NATS server < 2.10, filters might be ignored.
```

Comment:
there is a `connected_server_version` which can be used to check: https://github.com/nats-io/nats.py/blob/main/nats/aio/client.py#L1210C9-L1210C33

Reply:
this can't be invoked without a `nc` I suppose?
I checked the usage of `connected_server_version` and this is how its been used
`server_version = nc.connected_server_version`

Reply:
think it is part of the state already here, maybe via `self._js._nc`

Reply:
no,
I have tried that already 😄

Reply:
I see, thanks for checking